server/datastore/mysql/hosts.go

```diff
@@ -571,6 +584,8 @@ func (ds *Datastore) ListHosts(ctx context.Context, filter fleet.TeamFilter, opt
 }
 
 func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, filter fleet.TeamFilter, params []interface{}) (string, []interface{}) {
+	opt.OrderKey = defaultHostColumnTableAlias(opt.OrderKey)
```

Comment:
That's great to fix the specific cases of `created_at` and `updated_at`, but ultimately it would be best to validate that the order key is one that matches one of the fields available, and blank it (or return a 4xx error) if it's not, instead of accepting and using whatever gets provided. Not saying it should be part of this PR, but something to address as tech debt sooner than later IMO.

Reply:
I fully agree with that (and that's why I added the fixme comment), but whatever I add here to validate the data probably breaks something for someone. But we could add something like that with v5 maybe?

Reply:
Gotcha, I'm not sure validating to accept any of the available fields should cause any breaking change other than for requests that don't work as expected anyway? E.g. fields with a typo? I think considering this a breaking change and requiring v5 might be a stretch, given that it'd be fixing a buggy behaviour. We could validate in such a way that those using the `h.created_at` workaround would still work. @chiiph any thoughts about whether this should be a breaking change or not, or users taking advantage of the absence of validation beyond the `h.created_at` hack?

In any case, as mentioned I don't think it should block this PR and the fix (though we might need a new issue to track validating, if we decide to go forward with it), approving.